app/controllers/walks_controller.rb

```diff
@@ -41,7 +42,13 @@ def update
   def destroy
   end
 
-private
+  private
+
+  def verify_logged_in
```

Comment:
Awesome! If the other gating gems don't work out this is an easy and effective way to make this happen.